src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1105,7 +1105,8 @@ protected BoundExpression BindInferredVariableInitializer(DiagnosticBag diagnost
                     hasErrors = true;
                 }
             }
-            else if (kind == LocalDeclarationKind.Constant && initializerOpt != null && !localDiagnostics.HasAnyResolvedErrors())
+
+            if ((kind == LocalDeclarationKind.Constant || kind == LocalDeclarationKind.UsingVariable) && initializerOpt != null && !localDiagnostics.HasAnyResolvedErrors())
```

Comment:
>(kind == LocalDeclarationKind.Constant || kind == LocalDeclarationKind.UsingVariable) [](start = 16, length = 85)

This modified condition seems fragile.  Should this be `localSymbol.IsConst` ?

Reply:
If I understand the code there, `localSymbol.GetConstantValueDiagnostics(initializerOpt)` returns a nonempty bag only when `localSymbol.IsConst`.

---
In reply to: [393308709](https://github.com/dotnet/roslyn/pull/42481#discussion_r393308709) [](ancestors = 393308709)